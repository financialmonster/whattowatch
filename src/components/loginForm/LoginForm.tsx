import React, { FC } from 'react';
import { Formik, Form, Field, ErrorMessage } from 'formik';
import { useSelector } from 'react-redux';

import * as authSelectors from 'domains/auth/authSelectors';
import { loginFormShape } from 'formShapes';
import { TUserData } from 'types';

type TLoginFormProps = {
    loginFormSubmitHandler: (userData: TUserData) => void
}

export const LoginForm: FC<TLoginFormProps> = React.memo(({loginFormSubmitHandler}) => {
    const authError = useSelector(authSelectors.getAuthError);

    return (
        <Formik initialValues={loginFormShape.initialValues} onSubmit={loginFormSubmitHandler}
            validationSchema={loginFormShape.schema}
        >
            <Form className="sign-in__form">
                <ErrorMessage name="email">
                    {(message) => (
                        <div className="sign-in__message">
                            <p>{message}</p>
                        </div>
                    )}                   
                </ErrorMessage>
                <ErrorMessage name="password">
                    {(message) => (
                        <div className="sign-in__message">
                            <p>{message}</p>
                        </div>
                    )}
                </ErrorMessage>
                {(authError) &&
                    <div className="sign-in__message">
                        <p>We canâ€™t recognize this email <br/> and password combination. Please try again.</p>
                    </div>
                }
                <div className="sign-in__fields">
                    <div className="sign-in__field">
                        <Field className="sign-in__input" type="email" placeholder="Email address" name="email"
                            id="user-email"
                        />
                        <label className="sign-in__label visually-hidden" htmlFor="user-email">
                            Email address
                        </label>
                    </div>
                    <div className="sign-in__field">
                        <Field className="sign-in__input" type="password" placeholder="Password" name="password"
                            id="user-password"
                        />
                        <label className="sign-in__label visually-hidden" htmlFor="user-password">
                            Password
                        </label>
                    </div>
                </div>
                <div className="sign-in__submit">
                    <button className="sign-in__btn" type="submit" title="Sign in">
                        Sign in
                    </button>
                </div>
            </Form>
        </Formik>
    );
});
