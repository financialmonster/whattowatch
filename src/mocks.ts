import { Map, fromJS, List } from 'immutable';

export const createStoreMock = (mode: string = `common`) => {
    return {
        promo: Map({
            promo: Map(),
            isPromoFetching: true,
            promoError: null
        }),
        films: Map({
            films: fromJS([{genre: `Thriller`, id: 1}, {genre: `Comedy`, id: 2}]),
            isFilmsFetching: mode === `detailed`,
            filmsError: null,
            filter: `All genres`
        }),
        auth: Map({
            user: null,
            authError: null,
            isAuthFetching: false,
            authStatusError: null,
            isAuthStatusFetching: false
        }),
        reviews: Map({
            reviews: List(),
            isReviewsFetching: true,
            reviewsError: null
        })
    }
}
