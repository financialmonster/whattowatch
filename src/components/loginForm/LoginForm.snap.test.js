import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { LoginForm } from './LoginForm';
import { createStoreMock } from 'mocks';

it(`LoginForm: correctly rendered`, () => {
    const mockStore = configureStore();
    const store = mockStore(createStoreMock());

    const tree = renderer.create(
        <Provider store={store}>
            <LoginForm />
        </Provider>        
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
