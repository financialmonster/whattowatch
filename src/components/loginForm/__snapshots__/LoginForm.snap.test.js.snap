// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm: correctly rendered 1`] = `
<form
  action="#"
  className="sign-in__form"
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="sign-in__fields"
  >
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Email address"
        type="email"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-email"
      >
        Email address
      </label>
    </div>
    <div
      className="sign-in__field"
    >
      <input
        className="sign-in__input"
        id="user-password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value=""
      />
      <label
        className="sign-in__label visually-hidden"
        htmlFor="user-password"
      >
        Password
      </label>
    </div>
  </div>
  <div
    className="sign-in__submit"
  >
    <button
      className="sign-in__btn"
      disabled={false}
      title="Sign in"
      type="submit"
    >
      Sign in
    </button>
  </div>
</form>
`;
